index.bs

```diff
@@ -2153,6 +2158,82 @@ It is recommended that once [=explicit consent=] is granted for a specific [=/or
+
+Protected functionality {#protected-functionality}
+-----------------------
+
+The [=sensitive information=] exposed by the API can be divided into categories that share threat profiles and necessary protections against those threats.
+
+### Immersiveness ### {#protect-immersiveness}
+Users must be in control of when immersive sessions are created because the creation causes invasive changes on a user's machine. For example, starting an [=immersive session=] will engage the [=/XR device=] sensors, take over access to the device's display, and begin presentating immersive content which may terminate another application's access to the XR hardware. It may also incur significant power or performance overhead on some systems or trigger the launching of a status tray or storefront.
+
+<div class="algorithm" data-algorithm="immersive-session-allowed">
+
+To determine if an <dfn>immersive session request is allowed</dfn> the user agent MUST run the following steps:
+
+  1. If the request was not [=triggered by user activation=], return <code>false</code>
+  1. If the requesting document is not considered [=trustworthy=], return <code>false</code>
+  1. If [=user intent=] to begin an [=immersive session=] is not well understood, either via [=explicit consent=] or [=implicit consent=], return <code>false</code>
+  1. Return <code>true</code>
+
+</div>
+
+Starting an {{XRSessionMode/"inline"}} session does not implicitly carry the same requirements, though additional requirements may be imposed depending on the session's [=requested features=].
+
+<div class="algorithm" data-algorithm="inline-session-allowed">
+
+To determine if an <dfn>inline session request is allowed</dfn> the user agent MUST run the following steps:
+
+  1. If the requesting document is not [=responsible=], return <code>false</code>
+  1. If requesting document's origin is not allowed to use the WebXR [[#feature-policy|feature policy]] or the [=underlying sensors=]' feature policies, return <code>false</code>
+  1. Return <code>true</code>
+
+</div>
+
+### Poses ### {#protect-poses}
+When based on sensor data, {{XRPose}} and {{XRViewerPose}} will expose [=sensitive information=] that may be misused in a number of ways, including input sniffing, gaze tracking, or fingerprinting.
+
+<div class="algorithm" data-algorithm="poses-allowed">
+
+To determine if <dfn>poses may be reported</dfn> to an {{XRSession}} |session|, the user agent MUST run the following steps:
+
+  1. Let |document| be the document that owns |session|.
+  1. If the request does not originate from |document|, return <code>false</code>.
+  1. If |document| is not considered [=trustworthy=], return <code>false</code>.
+  1. If |session|'s {{XRSession/visibilityState}} in not {{XRVisibilityState/"visible"}}, return <code>false</code>.
```

Comment:
@toji sorry if I already asked this, should we also be checking be 'active and focused' ie:

```suggestion
  1. If |session|'s {{XRSession/visibilityState}} in not {{XRVisibilityState/"visible"}}, return <code>false</code>.
  1. If |document| is not [=active and focused=], return <code>false</code>.
```

Reply:
The `trustworthy` algorithm (referenced on line 2215 here) includes a check that the document is active and focused already.

Reply:
You're correct! I was looking at an older version of the algorithm. Thanks, you can resolve this.